### PR TITLE
test(conn-manager): watermark/scoring component 2

### PR DESCRIPTION
**Pull request by rlve**
## Summary

Extends the connection-manager component suite with cases for protection lifecycle, decaying scores, and the interaction between the watermark trim and the hard connection cap.

New cases:
- `unprotect` re-exposes a peer so a later trim can prune it
- Multi-tag protection holds until every tag is removed (`isProtected` flips only on the last `unprotect`)
- A decaying tag (`tagPeerDecaying` + `decayFixed`) keeps a peer until its score decays to zero, then the next trim drops it
- Trim stops short of `lowWater` when more than `lowWater` peers are protected, settling at the protected count
- Hard cap and watermark running together: trimmed connections release their semaphore slots, and available inbound slots settle at `cap - lowWater - 1` (accept loop keeps one slot pre-acquired)
- Hard cap rejects new connections when the trim has no prunable candidates: inbound `getIncomingSlot` blocks, outbound `getOutgoingSlot` raises `TooManyConnectionsError`
- Dial fails with `DialFailedError` when reaching `lowWater` forces the trim to drop the dialing peer's own freshly stored connection

## Affected Areas

- [x] Tests

## Impact on Library Users

None.

## Risk Assessment

None.

### CODE REVIEW

File: tests/libp2p/test_conn_manager_component.nim [hunk 1]
```diff
@@ -223,3 +223,237 @@ suite "Connection Manager Watermark/Scoring Component":
...
+      node.connManager.availableSlots(Direction.In) == 0
+
+    # with the cap full the semaphore rejects new connections
+    check not (await node.connManager.getIncomingSlot().withTimeout(100.millis))
+    expect TooManyConnectionsError:
+      discard node.connManager.getOutgoingSlot()
+
+  asyncTest "dial fails when the trim prunes the new connection":
+    # reaching lowWater can force the trim to drop the dialing peer's own freshly stored connection
+    # the dial then fails with DialFailedError instead of connecting and being trimmed cleanly later
```
Comment (rlve) [inline tests/libp2p/test_conn_manager_component.nim:433]: @vladopajic have a look on this case, is this acceptable or should I create an issue?

Comment (vladopajic) [reply tests/libp2p/test_conn_manager_component.nim:433]: ~~it works as expected; but questions is should it be improved~~

is this it: 
when node drops connection on its side, other node must also drop same connection (connection has to have to points). 
if this is is then it's fine

**Review by vladopajic — APPROVED** [review]

File: tests/libp2p/test_conn_manager_component.nim (hunk 1 above)
Comment (rlve) [reply tests/libp2p/test_conn_manager_component.nim:433]: I was rather thinking if this should be handled gracefully, connection allowed to happen and then trimmed? Anyway this is an edge case and happens because of `gracePeriod == 0`.

Comment (vladopajic) [reply tests/libp2p/test_conn_manager_component.nim:433]: that is fine.because all connections are scored when trimming.

**Review by gmelodie — APPROVED** [review]

**Review by richard-ramos — APPROVED** [review]
